pkg/util/kube/utils.go

```diff
@@ -140,13 +149,27 @@ func GetVolumeDirectory(pod *corev1api.Pod, volumeName string, pvcLister corev1l
 	}
 
 	// PV's been created with a CSI source.
-	if pv.Spec.CSI != nil {
+	if isProvisionedByCSI(pv) {
 		return pvc.Spec.VolumeName + "/mount", nil
 	}
 
 	return pvc.Spec.VolumeName, nil
 }
 
+func isProvisionedByCSI(pv *corev1api.PersistentVolume) bool {
+	if pv.Spec.CSI != nil {
+		return true
+	}
+	// Although the pv.Spec.CSI is nil, the volume could be provisioned by a CSI driver when enabling the CSI migration
+	// Refer to https://github.com/vmware-tanzu/velero/issues/4496 for more details
+	if pv.Annotations != nil {
+		if strings.Contains(pv.Annotations[KubeAnnDynamicallyProvisioned], "csi") {
```

Comment:
for safety how about checking `.csi.`?

Can we add a log saying `because the provisioned-by annotation is xxxx so I think this is provisioned by CSI`?

Reply:
There are some drivers named `csi.xxx` and `csi-xxx`: https://kubernetes-csi.github.io/docs/drivers.html